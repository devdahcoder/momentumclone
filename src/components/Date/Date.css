.date-top-left {
  flex: 1 1 auto;
  position: relative;
  display: flex;
  padding: 8px;
  min-height: 150px;
  align-items: stretch;
}

.date-top-left .date {
  text-align: right;
  box-sizing: inherit;
  position: relative;
  display: inline-block;
  transition: opacity 0.5s ease;
  order: 0;
}

.date-top-left .date .date-metric-item {
  vertical-align: top;
  position: relative;
  display: inline-block;
  border-radius: 2px;
  text-align: right;
  padding: 11px 10px 10px;
  cursor: pointer;
  order: 1;
  padding-right: 20px;
}

.date-top-left .date .date-metric-item .date-metric-stat {
  text-align: right;
  cursor: pointer;
  position: relative;
  font-size: 1.625rem;
  font-weight: 400;
  margin-right: -1px;
  white-space: nowrap;
  height: 30px;
  display: flex;
  flex-direction: row;
}

.date-top-left .date .date-metric-item .date-metric-stat .hide {
  text-align: right;
  cursor: pointer;
  font-weight: 400;
  white-space: nowrap;
  display: block;
  position: absolute;
  left: 100%;
  top: 5px;
  animation: fadeout 4s ease 4s 1 normal;
  animation-fill-mode: forwards;
  font-size: 55%;
  text-transform: uppercase;
}

.date-top-left .date .date-metric-item .metric-label {
  text-align: right;
  cursor: pointer;
  margin: 0;
  padding: 0 0 3px;
  overflow-wrap: break-word;
  transition: opacity .5s ease;
  word-break: break-word;
  vertical-align: top;
  display: inline-block;
  font-size: .8125rem;
  outline: 0;
}

.date-top-left .date .date-metric-item .metric-label .metric-label-name {
  text-align: right;
  cursor: pointer;
  overflow-wrap: break-word;
  word-break: break-word;
  font-size: .8125rem;
  box-sizing: inherit;
  opacity: .85;
}
/*# sourceMappingURL=date.css.map */