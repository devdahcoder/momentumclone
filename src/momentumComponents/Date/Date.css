.date-top-left {
  color: #fff;
  font-size: 100%;
  text-shadow: 0 1px 5px rgba(0, 0, 0, 0.1);
  user-select: none;
  flex: 1 1 auto;
  position: relative;
  display: flex;
  height: 150px;
  align-items: stretch;
}

.date-top-left .header-date-container {
  color: #fff;
  font-size: 100%;
  text-shadow: 0 1px 5px rgba(0, 0, 0, 0.1);
  user-select: none;
  position: relative;
  display: inline-block;
  transition: opacity 0.5s ease;
  vertical-align: top;
}

.date-top-left .header-date-container span {
  position: relative;
  display: inline-block;
  color: rgba(255, 255, 255, 0.85);
  font-size: 106.25%;
  font-weight: 500;
  line-height: 60px;
  transition: color .1s ease, text-shadow .1s ease, opacity 0.5s ease;
  cursor: pointer;
  padding: 0 38px;
  padding-left: 20px;
}
/*# sourceMappingURL=date.css.map */