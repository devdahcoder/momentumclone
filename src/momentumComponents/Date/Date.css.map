{
    "version": 3,
    "mappings": "AAAA,AAAA,cAAc,CAAC;EACX,KAAK,EAAE,IAAI;EACX,SAAS,EAAE,IAAI;EACf,WAAW,EAAE,CAAC,CAAC,GAAG,CAAC,GAAG,CAAC,kBAAiB;EACxC,WAAW,EAAE,IAAI;EACjB,IAAI,EAAE,QAAQ;EACd,QAAQ,EAAE,QAAQ;EAClB,OAAO,EAAE,IAAI;EAGb,MAAM,EAAE,KAAK;EACb,WAAW,EAAE,OAAO;CA2BvB;;AAtCD,AAaI,cAbU,CAaV,sBAAsB,CAAC;EACnB,KAAK,EAAE,IAAI;EACX,SAAS,EAAE,IAAI;EACf,WAAW,EAAE,CAAC,CAAC,GAAG,CAAC,GAAG,CAAC,kBAAiB;EACxC,WAAW,EAAE,IAAI;EACjB,QAAQ,EAAE,QAAQ;EAClB,OAAO,EAAE,YAAY;EACrB,UAAU,EAAE,iBAAiB;EAC7B,cAAc,EAAE,GAAG;CAgBtB;;AArCL,AAuBQ,cAvBM,CAaV,sBAAsB,CAUlB,IAAI,CAAC;EACD,QAAQ,EAAE,QAAQ;EAClB,OAAO,EAAE,YAAY;EACrB,KAAK,EAAE,yBAAuB;EAC9B,SAAS,EAAE,OAAO;EAClB,WAAW,EAAE,GAAG;EAChB,WAAW,EAAE,IAAI;EACjB,UAAU,EAAE,yDAEK;EACjB,MAAM,EAAE,OAAO;EACf,OAAO,EAAE,MAAM;EACf,YAAY,EAAE,IAAI;CACrB",
    "sources": [
        "Date.scss"
    ],
    "names": [],
    "file": "Date.css"
}