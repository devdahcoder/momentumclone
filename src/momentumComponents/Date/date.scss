.date-top-left {
    color: #fff;
    font-size: 100%;
    text-shadow: 0 1px 5px rgba(0, 0, 0, .1);
    user-select: none;
    flex: 1 1 auto;
    position: relative;
    display: flex;
    // --max-height: 60px;
    // --side-padding: 8px;
    height: 150px;
    align-items: stretch;
    display: flex;
    flex-direction: column;

    .header-date-container {
        color: #fff;
        font-size: 100%;
        text-shadow: 0 1px 5px rgba(0, 0, 0, .1);
        user-select: none;
        position: relative;
        display: inline-block;
        transition: opacity 0.5s ease;
        vertical-align: top;

        span {
            position: relative;
            display: inline-block;
            color: hsla(0, 0%, 100%, 0.85);
            font-size: 106.25%;
            font-weight: 500;
            line-height: 60px;
            transition: color .1s ease,
            text-shadow .1s ease,
            opacity 0.5s ease;
            cursor: pointer;
            padding: 0 38px;
            padding-left: 20px;
        }
    }
}