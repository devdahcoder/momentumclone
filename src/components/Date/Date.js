import React, {useEffect, useState, useContext} from 'react';
import "./Date.css";


// imported utils
import {DateContext} from "../../context/DateContext"

const Date = () => {

    const {currentDate} = useContext(DateContext);

    const {day, date} = currentDate;

    return (

        <div data-testid="date-element" className="date-top-left">

            <div data-testid="show-fade" className="app-container weather show show-fade-in">

                <div data-testid="data-metric-item-test" className="app-dash date-metric-item" title={date}>

                    <div className="date-metric-stat">

                        <div>

                            <span data-testid="date-unit" className="unit hide">{date}</span>

                        </div>

                    </div>
                    
                    <div className="location metric-label data" data-test="location" title={day}>

                        <span className="metric-label-name">{day}</span>

                    </div>

                    <span className="metric-message"></span>

                </div>

            </div>

        </div>

    )

}

export default Date