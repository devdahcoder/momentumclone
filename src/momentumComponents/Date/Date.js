import React from 'react';
import "./Date.css"


const Date = () => {
    return (
        <div className="region date-top-left">
            <div id="quicklinks" className="header-date-container quicklinks links" data-test="undefined" style={{}}>
                <span className="app-dash toggle Links-toggle" data-test="links-app-dash" data-ob="links-app-dash">23, 4, 2020</span>
                {/* <div class="app-wrapper app-placeholder nipple nipple-top-left">
                    <div class="app links-app" style={{height:"70px", width:"260px"}}>
                        <div class="app-placeholder-loading">
                            <i class="loading-icon"></i>Loading...
                        </div>
                    </div>
                </div> */}
            </div>
            <div id="dashlinks" class="dashlinks"></div>
        </div>
    )
}

export default Date