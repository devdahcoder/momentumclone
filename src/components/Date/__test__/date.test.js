import React from 'react';
import { render } from '@testing-library/react';
import Date from "../Date";
import DateContextProvider from "../../../context/DateContext";

test("should render in the document", () => {

    const { getByTestId } = render(<DateContextProvider><Date /></DateContextProvider>);

    const dateElement = getByTestId("date-element");

    expect(dateElement).toBeInTheDocument();

    expect(dateElement).toHaveClass("date-top-left");

    expect(dateElement).toMatchSnapshot();

})

test("should render in the document", () => {

    const { getByTestId } = render(<DateContextProvider><Date /></DateContextProvider>);

    const dateElement = getByTestId("show-fade");

    expect(dateElement).toBeInTheDocument();

    expect(dateElement).toHaveClass("app-container weather show show-fade-in");

})

test("should render in the document", () => {

    const { getByTestId } = render(<DateContextProvider><Date /></DateContextProvider>);

    const dateElement = getByTestId("data-metric-item-test");

    expect(dateElement).toBeInTheDocument();

    expect(dateElement).toHaveClass("app-dash date-metric-item");

})


test ("should render in the document", () => {
    
    const { getByTestId } = render(<DateContextProvider><Date /></DateContextProvider>);

    const dateElement = getByTestId("date-unit");

    expect(dateElement).toBeInTheDocument();

    expect(dateElement).toHaveClass("unit hide");
    
})
