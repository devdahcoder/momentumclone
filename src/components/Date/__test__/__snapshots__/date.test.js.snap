// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render in the document 1`] = `
<div
  class="date-top-left"
  data-testid="date-element"
>
  <div
    class="app-container weather show show-fade-in"
    data-testid="show-fade"
  >
    <div
      class="app-dash date-metric-item"
      data-testid="data-metric-item-test"
    >
      <div
        class="date-metric-stat"
      >
        <div>
          <span
            class="unit hide"
            data-testid="date-unit"
          />
        </div>
      </div>
      <div
        class="location metric-label data"
        data-test="location"
      >
        <span
          class="metric-label-name"
        />
      </div>
      <span
        class="metric-message"
      />
    </div>
  </div>
</div>
`;
